alfresco-health-processor-platform/src/test/java/eu/xenit/alfresco/healthprocessor/plugins/solr/SolrIndexValidationHealthProcessorPluginTest.java

```diff
@@ -0,0 +1,243 @@
+                String.format(SolrIndexValidationHealthProcessorPlugin.FMT_NOT_INDEXED_TX, searchEndpoint2)
+        ))), "Expect all nodes to have a message");
+    }
+
+    @Test
+    void process_no_endpoints() {
+        when(searchEndpointSelector.getSearchEndpointsForNode(Mockito.any())).thenReturn(Collections.emptySet());
+
+        when(nodeService.getNodeStatus(Mockito.any())).then(invocation -> {
+            NodeRef nodeRef = invocation.getArgument(0);
+            return new Status(1L, nodeRef, "1", 1L, false);
+        });
+
+        Set<NodeRef> nodeRefs = set(TestNodeRefs.REFS);
+        Set<NodeHealthReport> healthReports = healthProcessorPlugin.process(nodeRefs);
+
+        // Verify that no search endpoints were contacted
+        verifyNoInteractions(solrSearchExecutor);
+
+        assertEquals(nodeRefs.size(), healthReports.size(), "Expected an equal number of health reports as the number of passed noderefs");
+        assertTrue(healthReports.stream().map(NodeHealthReport::getStatus).allMatch(Predicate.isEqual(NodeHealthStatus.NONE)), "Expect all nodes to be none");
+        assertTrue(healthReports.stream().map(NodeHealthReport::getMessages).allMatch(Predicate.isEqual(set(
+               SolrIndexValidationHealthProcessorPlugin.MSG_NO_SEARCH_ENDPOINTS
+        ))), "Expect all nodes to have a message");
+    }
+
+    @Test
+    void process_multiple_endpoints_exception() throws IOException {
+        SearchEndpoint searchEndpoint1 = new SearchEndpoint(URI.create("http://empty/solr/index1/"));
+        SearchEndpoint searchEndpoint2 = new SearchEndpoint(URI.create("http://empty/solr/index2/"));
+        when(searchEndpointSelector.getSearchEndpointsForNode(Mockito.any())).thenReturn(set(searchEndpoint1, searchEndpoint2));
+
+        when(nodeService.getNodeStatus(Mockito.any())).then(invocation -> {
+            NodeRef nodeRef = invocation.getArgument(0);
+            return new Status(1L, nodeRef, "1", 1L, false);
+        });
+
+        // All endpoint 1 throws an exception, endpoint 2 says they are present
+        when(solrSearchExecutor.checkNodeIndexed(Mockito.eq(searchEndpoint1), Mockito.any())).thenThrow(new IOException("My server is broken"));
+        when(solrSearchExecutor.checkNodeIndexed(Mockito.eq(searchEndpoint2), Mockito.any())).thenAnswer(invocation -> new SolrSearchResult(invocation.getArgument(1, Set.class), Collections.emptySet(), Collections.emptySet()));
+
+        Set<NodeRef> nodeRefs = set(TestNodeRefs.REFS);
+        Set<NodeHealthReport> healthReports = healthProcessorPlugin.process(nodeRefs);
+
+        // Verify that each search endpoint was only called *once*
+        verify(solrSearchExecutor).checkNodeIndexed(Mockito.eq(searchEndpoint1), Mockito.any());
+        verify(solrSearchExecutor).checkNodeIndexed(Mockito.eq(searchEndpoint2), Mockito.any());
+
+        assertEquals(nodeRefs.size(), healthReports.size(), "Expected an equal number of health reports as the number of passed noderefs");
```

Comment:
If putting it in a separate function, it would be a function with many parameters.
After each is not possible because these are local variables.

So I am not sure that it would reduce complexity, instead of making it harder to see what is actually asserted because it is hidden behind a function call.